Firestore/core/test/unit/model/value_util_test.cc

```diff
@@ -134,12 +134,18 @@ class ValueUtilTest : public ::testing::Test {
   }
 
   void VerifyDeepClone(const google_firestore_v1_Value& value) {
-    // `Message` takes ownership of the proto value and destroys it. We use
-    // it to verify that all values own their own memory.
-    nanopb::Message<google_firestore_v1_Value> clone1{DeepClone(value)};
-    nanopb::Message<google_firestore_v1_Value> clone2{DeepClone(*clone1)};
-    EXPECT_TRUE(value == *clone1);
-    EXPECT_TRUE(value == *clone2);
+    nanopb::Message<google_firestore_v1_Value> clone1;
+
+    [&] {
+      nanopb::Message<google_firestore_v1_Value> clone2{DeepClone(value)};
+      EXPECT_TRUE(value == *clone2)
```

Comment:
You can use `EXPECT_EQ`. Also, a failure message should contain the line where it happened, so the before/after message is probably not necessary.

Reply:
I wasn't able to get this to work:

```
/Users/mrschmidt/GitHub/firebase/firebase-ios-sdk/cmake-build-debug/external/src/googletest/googletest/include/gtest/gtest.h:1527:11: error: invalid operands to binary expression ('const firebase::firestore::_google_firestore_v1_Value' and 'const firebase::firestore::_google_firestore_v1_Value')
  if (lhs == rhs) {
      ~~~ ^  ~~~
/Users/mrschmidt/GitHub/firebase/firebase-ios-sdk/cmake-build-debug/external/src/googletest/googletest/include/gtest/gtest.h:1554:12: note: in instantiation of function template specialization 'testing::internal::CmpHelperEQ<firebase::firestore::_google_firestore_v1_Value, firebase::firestore::_google_firestore_v1_Value>' requested here
    return CmpHelperEQ(lhs_expression, rhs_expression, lhs, rhs);
           ^
/Users/mrschmidt/GitHub/firebase/firebase-ios-sdk/Firestore/core/test/unit/model/value_util_test.cc:141:7: note: in instantiation of function template specialization 'testing::internal::EqHelper::Compare<firebase::firestore::_google_firestore_v1_Value, firebase::firestore::_google_firestore_v1_Value, nullptr>' requested here
      EXPECT_EQ(value, *clone2)
      ^
/Users/mrschmidt/GitHub/firebase/firebase-ios-sdk/cmake-build-debug/external/src/googletest/googletest/include/gtest/gtest.h:2028:54: note: expanded from macro 'EXPECT_EQ'
  EXPECT_PRED_FORMAT2(::testing::internal::EqHelper::Compare, val1, val2)
                                                     ^
/Users/mrschmidt/GitHub/firebase/firebase-ios-sdk/Firestore/core/include/firebase/firestore/geo_point.h:113:13: note: candidate function not viable: no known conversion from 'const firebase::firestore::_google_firestore_v1_Value' to 'const firebase::firestore::GeoPoint' for 1st argument
inline bool operator==(const GeoPoint& lhs, const GeoPoint& rhs) {
            ^
/Users/mrschmidt/GitHub/firebase/firebase-ios-sdk/cmake-build-debug/external/src/googletest/googletest/include/gtest/gtest.h:1518:13: note: candidate function not viable: no known conversion from 'const firebase::firestore::_google_firestore_v1_Value' to 'testing::internal::faketype' for 1st argument
inline bool operator==(faketype, faketype) { return true; }
```

I tried to use EXPECT_EQ in my earlier attempts at writing these tests and for some reason the `==` operator is not recognized.

Reply:
Ok, what happens is that due to the way `EXPECT_EQ` is implemented, the `operator==` has to be in the same namespace as at least one of the classes being compared. In this case, `google_firestore_v1_Value` is in the `firebase::firestore` namespace, but the comparison operator is in `firebase::firestore::model`. It's probably a good idea to move `operator==` one namespace level up.